### PR TITLE
Remove unused IPython.utils.rlineimpl readline module

### PR DESCRIPTION
**Pull request by takluyver**
Also deprecate the use_readline parameter for IPCompleter

### CODE REVIEW

**Comment by takluyver** [comment]
There's a probably unrelated error on Python nightly; I'm guessing it's a broken build on Travis at the moment.

```
  File "/home/travis/virtualenv/python3.6-dev/lib/python3.6/re.py", line 122, in <module>
    import enum
ModuleNotFoundError: No module named 'enum'
```

File: IPython/core/completer.py [hunk 1]
```diff
@@ -608,20 +609,15 @@ def __init__(self, shell=None, namespace=None, global_namespace=None,
...
         self.splitter = CompletionSplitter()
 
-        # Readline configuration, only used by the rlcompleter method.
         if use_readline:
-            # We store the right version of readline so that later code
-            import IPython.utils.rlineimpl as readline
-            self.readline = readline
-        else:
-            self.readline = None
+            warnings.warn('The use_readline parameter is deprecated since IPython 6.0.',
```
Comment (willingc) [inline IPython/core/completer.py]: Perhaps nitpicky:

``'The use_readline parameter is deprecated and ignored since IPython 6.0.'``

**Comment by willingc** [comment]
Looks good to merge @talkluyver. One small comment on the warning message.

File: IPython/core/completer.py (hunk 1 above)
Comment (takluyver) [inline IPython/core/completer.py]: Thanks, fixed.

**Comment by willingc** [comment]
Thanks @takluyver

**Pull request merged**